google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -971,13 +971,15 @@ private Boolean retryOnRetryableError(Code errorCode, AppendRequestAndResponse r
       try {
         requestWrapper.retryCount++;
         if (this.retrySettings != null && errorCode == Code.RESOURCE_EXHAUSTED) {
-          // Trigger exponential backoff in append loop when request is resent for quota errors
-          if (requestWrapper.attemptSettings == null) {
-            requestWrapper.attemptSettings = requestWrapper.retryAlgorithm.createFirstAttempt();
-          } else {
-            requestWrapper.attemptSettings =
-                requestWrapper.retryAlgorithm.createNextAttempt(requestWrapper.attemptSettings);
-          }
+          // Trigger exponential backoff in append loop when request is resent for quota errors.
+          // createNextAttempt correctly initiales the retry delay; createfirstAttempt does not
+          // include a positive delay, just 0.
+          requestWrapper.attemptSettings =
+              requestWrapper.attemptSettings == null
+                  ? requestWrapper.retryAlgorithm.createFirstAttempt()
+                  : requestWrapper.attemptSettings;
+          requestWrapper.attemptSettings =
```

Comment:
this line can be removed

Reply:
What I meant to say is that lines 977 - 982 can be replaced with:

requestWrapper.attemptSettings = requestWrapper.retryAlgorithm.createNextAttempt(
requestWrapper.attemptSettings == null
                  ? requestWrapper.retryAlgorithm.createFirstAttempt()
                  : requestWrapper.attemptSettings
);

Reply:
Fixed